Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -314,7 +362,9 @@ protected override DbTransaction BeginDbTransaction(IsolationLevel isolationLeve
 
         protected override DbCommand CreateDbCommand()
         {
-            return new SnowflakeDbCommand(this);
+            var command = DbProviderFactory.CreateCommand();

```

Comment:
what is the relation of this code change with the connection pool change?

Reply:
allows to inject a different objects factory in the tests to provide mocked commands and test edge case scenarios with exceptions thrown during rollback of transaction